openapi.yaml

```diff
@@ -1,279 +1,720 @@
-swagger: '2.0'
+openapi: 3.0.0
 info:
-  title: Express API Skeleton
-  description: Skeleton for Express APIs
-  version: v1
+  title: Locations API
+  description: Get information about OSU locations and services
+  version: v2
   license:
     name: GNU Affero General Public License Version 3
     url: http://www.gnu.org/licenses/agpl-3.0.en.html
   contact:
     name: IS Data Architecture Team
     url: https://is.oregonstate.edu/data-architecture
     email: isdataarchitecture@oregonstate.edu
-schemes:
-  - https
-host: api.oregonstate.edu
-basePath: /v1
 externalDocs:
   description: GitHub Repository
-  url: https://github.com/osu-mist/express-api-skeleton
-produces:
-  - application/json
-consumes:
-  - application/json
+  url: https://github.com/osu-mist/locations-frontend-api
+servers:
+  - url: https://api.oregonstate.edu/v2
 security:
   - OAuth2:
       - full
 paths:
-  /pets:
+  /locations:
     get:
-      summary: List all pets
       tags:
-        - pets
-      description: Get all pets
-      operationId: getPets
+        - locations
+      summary: Get locations
+      description: Gets all `Location` objects. Can be filtered by parameters.
+      operationId: getLocations
       parameters:
-        - $ref: '#/parameters/pageNumber'
-        - $ref: '#/parameters/pageSize'
         - in: query
-          name: species
-          description: Search by pet species
+          name: filter[search]
+          description: Search through buildings based on building name and abbreviation
           required: false
-          type: string
+          schema:
+            type: string
+        - in: query
+          name: filter[name]
+          description: Search buildings by name
+          required: false
+          schema:
+            type: string
```

Comment:
added a `filter[name]` query to search specifically for buildings by name instead of combined name and abbreviation in `search`

Reply:
Adding `filter[name][fuzzy]` to allow fuzzy searching of a location's name